packages/@aws-cdk/aws-vpcv2-alpha/lib/ipam.ts

```diff
@@ -0,0 +1,537 @@
+   * The locale (AWS Region) of the pool. Should be one of the IPAM operating region.
+   *  Only resources in the same Region as the locale of the pool can get IP address allocations from the pool.
+   * You can only allocate a CIDR for a VPC, for example, from an IPAM pool that shares a locale with the VPC’s Region.
+   * Note that once you choose a Locale for a pool, you cannot modify it. If you choose an AWS Region for locale that has not been configured as an operating Region for the IPAM, you'll get an error.
+   * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html#cfn-ec2-ipampool-locale
+   * @default - Current operating region
+   */
+  readonly locale?: string;
+
+  /**
+   * The IP address source for pools in the public scope.
+   * Only used for IPv6 address
+   * Only allowed values to this are 'byoip' or 'amazon'
+   * @default amazon
+   */
+  readonly publicIpSource?: IpamPoolPublicIpSource;
+
+  /**
+  * Limits which service in AWS that the pool can be used in.
+  *
+  * "ec2", for example, allows users to use space for Elastic IP addresses and VPCs.
+  *
+  * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html#cfn-ec2-ipampool-awsservice
+  *
+  * @default - No service
+  */
+  readonly awsService?: AwsServiceName;
+}
+
+/**
+ * Properties for creating an IPAM pool.
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html
+ */
+interface IpamPoolProps extends PoolOptions {
+  /**
+   * Scope id where pool needs to be created
+   */
+  readonly ipamScopeId: string;
+
+  /**
+   * IPAM resource name
+   * @default autogenerated by CDK
+   */
+  readonly ipamPoolName?: string;
+}
+
+/**
+ * Options to provision CIDRs to an IPAM pool.
+ * Used to create a new IpamPoolCidr
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampoolcidr.html
+ */
+export interface IpamPoolCidrProvisioningOptions {
+  /**
+   * Ipv6 Netmask length for the CIDR
+   * @default none
+   */
+  readonly netmaskLength?: number;
+
+  /**
+   * Ipv6 CIDR block for the IPAM pool
+   * @default none
+   */
+  readonly cidr?: string;
+}
+
+/**
+ * Definition used to add or create a new IPAM pool
+ */
+export interface IIpamPool{
+  /**
+ * Pool ID to be passed to the VPC construct
+ * @attribute IpamPoolId
+ */
+  readonly ipamPoolId: string;
+
+  /**
+   * Pool CIDR for IPv6 to be provisioned with Public IP source set to 'Amazon'
+   */
+  readonly ipamCidrs: CfnIPAMPoolCidr[];
+
+  /**
+   * Function to associate a IPv6 address with IPAM pool
+   */
+  provisionCidr(id: string, options: IpamPoolCidrProvisioningOptions): CfnIPAMPoolCidr;
+
+}
+
+/**
+ * IPAM scope is the highest-level container within IPAM. An IPAM contains two default scopes.
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipamscope.html
+ */
+export interface IpamScopeProps {
```

Comment:
[blocking] should this interface extends `IpamScopeOptions`?

Reply:
if we do it we kind of open all possible values to the customer, including providing scope id. 
This is only for the values that customer should be able to provide using addScope() method in IPAM.